### PR TITLE
chore: Add ESLint config

### PR DESCRIPTION
**Pull request by nschonni**
Was going to look at splitting the new link rule, and realized that now that we have "real" JS here, linting is probably a good idea.

### CODE REVIEW

**Comment by Trott** [comment]
With the current code, `npm test` succeeds if `tmp` doesn't exist. With this code, the user gets an error.

```
tmp/test/README.md
  1:1  error  No such file or directory
```

I'm OK with that, even think it might be an improvement on some level, but I wonder if we might want to add a check for `tmp` existence and output a link to a doc explaining what to do. (No `node` checkout on your file system? Provide instructions to checkout `node` into `tmp`. Have a node checkout? Provide instructions to make a symlink on non-Windows systems and instructions for whatever the equivalent would be on Windows systems.)

Not necessary to do it in this PR, or even ever since this tool is maintained by a very small group of people. But if someone wants to make the effort to add that documentation to the README.md file, awesome.

**Review by Trott — APPROVED** [review]
LGTM although non-blocking nits on the files that don't have newlines at the end `.eslintignore` and `.prettierignore` and my previous non-blocking nit on the error message if there's no `tmp` directory.

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+tmp/
```
Comment (aduh95) [inline .eslintignore:1]:
```suggestion
tmp/

```

File: .prettierignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+tmp/
+*.md
```
Comment (aduh95) [inline .prettierignore:2]:
```suggestion
*.md

```

**Review by aduh95 — APPROVED** [review]

File: package-lock.json [hunk 3]
```diff
@@ -1,2138 +1,8 @@
 {
   "name": "remark-preset-lint-node",
   "version": "2.4.1",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
Comment (nschonni) [inline package-lock.json:4]: Sorry for this noise, but I use Node 14 + NPM 6 still

**Comment by nschonni** [comment]
I added in EditorConfig and fixed those missing EOL @aduh95

**Comment by Trott** [comment]
I'm good with this to land as-is, if you are @aduh95.

**Review by aduh95 — APPROVED** [review]

**Pull request merged**